upload_vehicle_data.py

```diff
@@ -0,0 +1,327 @@
+"""This script automates the upload of rosbags from vehicle to VDI machine."""
+
+import argparse
+import logging
+import os
+import subprocess
+
+# Setup logging
+logging.basicConfig(
+    filename='upload_vehicle_data.log',
+    level=logging.INFO,
+    format='%(asctime)s %(levelname)s: %(message)s',
+)
+
+
+def run_ssh_command(remote_user, remote_ip, password, command):
+    """Run a command on the remote machine using SSH."""
+    ssh_command = (
+        f"sshpass -p {password} ssh {remote_user}@{remote_ip} '{command}'"
+    )
+    try:
+        subprocess.run(ssh_command, shell=True, check=True)
+        logging.info(f'Ran SSH command: {command}')
+    except subprocess.CalledProcessError as e:
+        logging.error(f'Failed to run SSH command: {command}: {e}')
+        raise
+
+
+def create_remote_directory(
+    remote_user, remote_ip, remote_temp_directory, password
+):
+    """Create a directory on the remote machine."""
+    command = f'mkdir -p {remote_temp_directory}'
+    run_ssh_command(remote_user, remote_ip, password, command)
+
+
+def delete_remote_directory(
+    remote_user, remote_ip, remote_temp_directory, password
+):
+    """Delete a directory on the remote machine."""
+    command = f'rm -rf {remote_temp_directory}'
+    run_ssh_command(remote_user, remote_ip, password, command)
+
+
+def compress_and_transfer_rosbag(
+    remote_user,
+    remote_ip,
+    rosbag_path,
+    remote_temp_directory,
+    vdi_upload_directory,
+    password,
+):
+    """Compress rosbag on remote machine and transfer it to VDI machine."""
+    # Compress the rosbag on the remote machine
+    remote_compressed_path = os.path.join(
+        remote_temp_directory, os.path.basename(rosbag_path)
+    )
+    mcap_path = '/home/linuxbrew/.linuxbrew/bin/mcap'

```

Comment:
Where does that `linuxbrew` comes from?

I think we need to parametrise this. Ideally, this should check first if the `mcap` cli library is installed or not.

Then in our README  "Vehicle server Set Up" section we should mention  `mcap` library needs to be installed and that the script will need to know where exactly.

Alternatively, we can mention steps to install `mcap` binary and then add it to the `PATH` env of the vehicle server so the command can be called from anywhere

Reply:
The problem is that when calling mcap directly on the rhaaegal machine the mcap command is include in the path, but when calling it remotely like `shikai@eidf150-main:~$ ssh shikai@129.215.117.104 mcap` we will get `bash: line 1: mcap: command not found`. When we do like `shikai@eidf150-main:~$ ssh shikai@129.215.117.104 /home/linuxbrew/.linuxbrew/bin/mcap` there is no preblem.

Reply:
How do you installed `mcap`?  Where does that `/home/linuxbrew` comes from? Is that a new user?

Reply:
Okay, I can see that was `homebrew` linux installation. I will suggest adding in the README a way to install `mcap` withouth this dep:

```
wget -O $HOME/mcap https://github.com/foxglove/mcap/releases/download/releases%2Fmcap-cli%2Fv0.0.47/mcap-linux-amd64 && chmod +x mcap
```
And in the script assume that the binary is in `$HOME/mcap`

Reply:
added and changed